### PR TITLE
feat(provider): add optional SSH port param to node in provider ssh block

### PR DESCRIPTION
**Pull request by rafsaf**
Hi, there are 2 use cases for having custom port in node block (rather than currently hard-coded port 22)
- it is quite natural to have more than one node behind single IP address, eg. using router with port forwarding or some iptables forward rules from one node to others
- sometimes people change ssh port for "security" purposes to some non standard number, this is also common requirement in some corporations ;)

PS. this is my first go code ever, so this may not be perfect (eg `//nolint: nestif`, no idea how to please linter warning there) and maybe I didn't completely get it, but tried it with some 22222 port on make example and probably works totally fine.

### Contributor's Note
Please mark the following items with an [x] if they apply to your PR.
Leave the [ ] if they are not applicable, or if you have not completed the item.
- [x] I have added / updated documentation in `/docs` for any user-facing features or additions.
- [ ] I have added / updated templates in `/examples` for any new or updated resources / data sources.
- [x] I have ran `make examples` to verify that the change works as expected. 

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by bpg** [comment]
@Hey @rafsaf! 👋🏼

Thank you so much for contribution, this is a good improvement!
There are few bugs that I'd like to merge & release first, then I go back to your PR.

Will keep in touch 🙂

File: proxmoxtf/provider/provider.go [hunk 1]
```diff
@@ -129,12 +129,17 @@ func providerConfigure(_ context.Context, d *schema.ResourceData) (interface{},
...
-	nodeOverrides := map[string]string{}
+	nodeOverrides := map[string]ssh.ProxmoxNode{}
 
 	if ns, ok := sshConf[mkProviderSSHNode]; ok {
 		for _, n := range ns.([]interface{}) {
 			node := n.(map[string]interface{})
-			nodeOverrides[node[mkProviderSSHNodeName].(string)] = node[mkProviderSSHNodeAddress].(string)
+			ProxmoxNode := ssh.ProxmoxNode{
+				Address: node[mkProviderSSHNodeAddress].(string),
+				Port:    int32(node[mkProviderSSHNodePort].(int)),
```
Comment (bpg-dev) [inline proxmoxtf/provider/provider.go]: really minor, to simply a bit:
```suggestion
				Port:    node[mkProviderSSHNodePort].(int32),
```

File: proxmoxtf/provider/provider.go [hunk 2]
```diff
@@ -129,12 +129,17 @@ func providerConfigure(_ context.Context, d *schema.ResourceData) (interface{},
...
 	}
 
-	nodeOverrides := map[string]string{}
+	nodeOverrides := map[string]ssh.ProxmoxNode{}
 
 	if ns, ok := sshConf[mkProviderSSHNode]; ok {
 		for _, n := range ns.([]interface{}) {
 			node := n.(map[string]interface{})
-			nodeOverrides[node[mkProviderSSHNodeName].(string)] = node[mkProviderSSHNodeAddress].(string)
+			ProxmoxNode := ssh.ProxmoxNode{
```
Comment (bpg-dev) [inline proxmoxtf/provider/provider.go]: this var doesn't need to be exported (i.e. with capital `P`)

**Review by bpg — APPROVED** [review]
Thanks for the feature @rafsaf!

LGTM! 🚀

**Comment by bpg** [comment]
@all-contributors please add @rafsaf for code

**Comment by rafsaf** [comment]
Good to see this merged @bpg. I needed this feature, thanks for looking at it.

**Pull request merged**